zinit-additional.zsh

```diff
@@ -101,55 +100,120 @@ function $f {
+    if [[ $ZINIT[DTRACE] = 1 ]]; then
+        +zi-log "{e} Debug mode currently active"
         return 1
     fi
-
-    ZINIT[DTRACE]=1
-
-    .zinit-diff _dtrace/_dtrace begin
-
-    # Full shadeing on
-    .zinit-tmp-subst-on dtrace
+    (){
+        ZINIT[DTRACE]=1
+        .zinit-diff _dtrace/_dtrace begin
+        .zinit-tmp-subst-on dtrace
+    }
+    if (( $? == 0 )); then
+        +zi-log "{i} Started debug mode"
+        return 0
+    fi
+} # ]]]
+# FUNCTION: .zinit-debug-status [[[
+# Revert changes made during debug mode
+.zinit-debug-status() {
+    +zi-log -n '{m} Debug mode: '
+    (( ZINIT[DTRACE] )) && +zi-log 'running' || +zi-log 'stopped'
 } # ]]]
 # FUNCTION: .zinit-debug-stop [[[
-# Stops Dtrace (i.e., session tracking for changes in Zsh state).
+# Stop debug mode
 .zinit-debug-stop() {
-    ZINIT[DTRACE]=0
-
-    # Shadowing fully off
-    .zinit-tmp-subst-off dtrace
-
-    # Gather end data now, for diffing later
-    .zinit-diff _dtrace/_dtrace end
-} # ]]]
-# FUNCTION: .zinit-clear-debug-report [[[
-# Forgets dtrace repport gathered up to this moment.
-.zinit-clear-debug-report() {
-    .zinit-clear-report-for _dtrace/_dtrace
+    if [[ $ZINIT[DTRACE] = 0 ]]; then
+        +zi-log '{e} Debug mode not active'
+        return 0
+    else
+        ZINIT[DTRACE]=0
+        (){
+            .zinit-tmp-subst-off dtrace     # turn of shadowing
+            .zinit-diff _dtrace/_dtrace end # get end data to diff later
+        }
+        if (( $? == 0 )); then
+            +zi-log '{i} Stopped debug mode'
+            return 0
+        fi
+    fi
 } # ]]]
-# FUNCTION: .zinit-debug-unload [[[
-# Reverts changes detected by dtrace run.
-.zinit-debug-unload() {
-    (( ${+functions[.zinit-unload]} )) || builtin source "${ZINIT[BIN_DIR]}/zinit-autoload.zsh" || return 1
-    if [[ ${ZINIT[DTRACE]} = 1 ]]; then
-        +zinit-message "{error}Dtrace is still active, stop it first with \`dstop'{rst}"
+# FUNCTION: +zinit-debug [[[
+# Debug command entry point
++zinit-debug(){
+    emulate -LR zsh ${=${options[xtrace]:#off}:+-o xtrace}
+    setopt extendedglob noksharrays nokshglob nullglob typesetsilent warncreateglobal
+    local o_help cmd
+    local -a usage=(
+      'Usage:'
+      '  zinit debug [options] [command]'
+      ' '
+      'Options:'
+      '  -h, --help     Show list of command-line options'
+      ' '
+      'Commands:'
+      '  clear     Clear debug report'
+      '  report    Show debug report'
+      '  revert    Revert changes made during debug mode'
+      '  start     Start debug mode'
+      '  status    Current debug status'
+      '  stop      Stop debug mode'
+    )
+    zmodload zsh/zutil
+    zparseopts -D -F -K -- \
+        {h,-help}=o_help \
+    || return 1
+
+    (( $#o_help )) && {
+        print -l -- $usage
+        return 0
+    }
+
+    cmd="$1"
+    if (( ! $+functions[.zinit-debug-${cmd}] )); then
+        print -l -- $usage
+        return 1
```

Comment:
Changes:
- logically group the debug sub-commands
- add `status` command
- add conditional log messages